src/mongocxx/validation_criteria.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/document/view.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+///
+/// Class representing criteria for document validation, to be applied to a collection.
+///
+/// TODO: add a link to documentation once it is published.
+///
+class MONGOCXX_API validation_criteria {
+   public:
+    ///
+    /// Set a validation rule for this validation object.
+    ///
+    /// @param rule
+    ///   Document representing a validation rule.
+    ///
+    void rule(bsoncxx::document::value rule);
+
+    ///
+    /// Set a validation rule using a document view. The given view must be of a
+    /// document::value that outlives this validation_criteria object.
+    ///
+    /// @param rule
+    ///   Document view representing a validation rule.
+    ///
+    void rule(bsoncxx::document::view rule);
+
+    ///
+    /// Set a validation level.
+    ///
+    /// @param level
+    ///   A validation level, "off," "strict," or "moderate."
+    ///
+    void level(stdx::string_view level);
```

Comment:
+1 for making this and the other enums, but if we decide not to, maybe this would be better as a `std::string`? Let's discuss when we talk about views / values